test/test_torch.py

```diff
@@ -2123,12 +2123,18 @@ def seed(generator):
             self.assertTrue((res1 >= 0).all().item())
 
     def test_randn(self):
-        torch.manual_seed(123456)
-        res1 = torch.randn(SIZE, SIZE)
-        res2 = torch.Tensor()
-        torch.manual_seed(123456)
-        torch.randn(SIZE, SIZE, out=res2)
-        self.assertEqual(res1, res2)
+        def common_routine(dtype):
+            torch.manual_seed(123456)
+            res1 = torch.randn(SIZE, SIZE, dtype=dtype)
+            res2 = torch.tensor([], dtype=dtype)
+            torch.manual_seed(123456)
+            torch.randn(SIZE, SIZE, out=res2)
+            self.assertEqual(res1, res2)
+
+        common_routine(dtype=torch.float32)
+        common_routine(dtype=torch.float64)
+        common_routine(dtype=torch.complex64)
+        common_routine(dtype=torch.complex128)
```

Comment:
There's dtype-generic testing framework in `test_torch.py` which is worth looking into, doesn't have to be for this PR though; see `@dtypes`

Reply:
We don't have a way of computing the variance on complex numbers yet (I think), but once this happens, it would be a good way to sanity check the statistical properties of `normal_`